scripts/ableplayer-base.js

```diff
@@ -38,14 +43,15 @@
 var AblePlayerInstances = [];
 
 (function ($) {
-	$(document).ready(function () {
-
-		$('video, audio').each(function (index, element) {
-			if ($(element).data('able-player') !== undefined) {
-				AblePlayerInstances.push(new AblePlayer($(this),$(element)));
-			}
-		});
-	});
+	// Initialise all AblePlayerInstances - this can be called
+	// by the React component that imports ableplayer
+  window.getAllAblePlayers = function() {
```

Comment:
This works for one react component, but what if there are two react components or instances of AblePlayer? The first one that loads will probably initialise AblePlayer elsewhere, and potentially some will get initialised multiple times.

What about instead exporting `AblePlayer`, and leaving it to the user (the react component) to call `new AblePlayer(element)`? That also removes this confusion in https://github.com/boronia/ableplayer/pull/1/files#r397577224, because the export is immutable (mutable exports work, but is very non-idionamic)

Reply:
So I tried that before by just importing `window.AblePlayer`, but ran into issues where I kept getting errors on `Ableplayer.prototype.xxx method not defined`, because for some reason, that constructor calls the prototype method while the instance is not created yet...? Makes no sense at all 😓  so this is the only way to make sure all the methods are defined in the right order before attempting to create the instances. 

I may have to show you the errors in a zoom session what I meant, it's quite hard to explain 😅

Reply:
You might have to still define `window.AblePlayer` initially (so that all the prototype methods can be attached), but then also set the export like `module.exports = window.AblePlayer;`.

Reply:
yeah, hapy to help out

Reply:
did it! 💪